tests/redirect.rs

```diff
@@ -116,13 +116,12 @@ async fn test_redirect_307_and_308_tries_to_post_again() {
         });
 
         let url = format!("http://{}/{}", redirect.addr(), code);
-        let dst = format!("http://{}/{}", redirect.addr(), "dst");
         let res = client.post(&url).body("Hello").send().await.unwrap();
-        assert_eq!(res.url().as_str(), dst);
-        assert_eq!(res.status(), reqwest::StatusCode::OK);
+        assert_eq!(res.url().as_str(), url);
+        assert_eq!(res.status(), code);
         assert_eq!(
             res.headers().get(reqwest::header::SERVER).unwrap(),
-            &"test-dst"
+            &"test-redirect"
```

Comment:
`test_redirect_307_and_308_tries_to_post_again`  won't pass since `tower_http::follow_redirect::FollowRedirect` doesn't redirect the 307 and 308 HTTP code.

Reply:
Interesting catch! And actually, this is precisely one of the main goals of the initiative: to improve the more general middleware with things that reqwest has learned. I think a PR to tower-http to understand those status codes would be a worthy improvement.

Reply:
>  tower_http::follow_redirect::FollowRedirect doesn't redirect the 307 and 308 HTTP code.

Hmm,  My last conclusiton isn't correct 😅 .  they don't redirect 307 and 308 code only if the `Body` is None.

So the `clone_body`  needs to be implemented otherwise it returns `None` by default.